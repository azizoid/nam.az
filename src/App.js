import React, { useState, useEffect } from "react";

import NavBar from "./Prayer/NavBar";
import Location from "./Prayer/Location";
import Progress from "./Prayer/Progress";
import PrayerList from "./Prayer/PrayerList";
import PrayerListStill from "./Prayer/PrayerListStill";
import Loader from "./Prayer/Loader";
import Ayah from "./Prayer/Ayah";

import {
  format,
  formatDistanceStrict,
  parse,
  differenceInSeconds,
  getDayOfYear,
} from "date-fns";
import { az } from "date-fns/locale";

const App = () => {
  const cities = [
    ,
    "Bakı",
    "Sumqayıt",
    "Gəncə",
    "Lənkaran",
    "Mingəçevir",
    "Naxçıvan",
    "Ağdam",
    "Astara",
    "Beyləqan",
    "Cəlilabad",
    "Göyçay",
    "Qəbələ",
    "Qazax",
    "Quba",
    "Qusar",
    "Saatlı",
    "Sabirabad",
    "Şamaxı",
    "Şəki",
    "Şirvan",
    "Xaçmaz",
    "Yevlax",
    "Zaqatala",
    "Xankəndi",
    // "Marneuli",
    // "Ərdəbil",
    // "Bolnisi",
  ];

  const [prayers, setPrayers] = useState([
    { id: 1, title: "Fəcr namazı", time: "--:--", rakat: 2 },
    { id: 2, title: "Günəş", time: "-:-", rakat: 0 },
    { id: 3, title: "Zöhr namazı", time: "-:-", rakat: 4 },
    { id: 4, title: "Əsr namazı", time: "-:-", rakat: 4 },
    { id: 5, title: "Məğrib namazı", time: "-:-", rakat: 3 },
    { id: 6, title: "İşa namazı", time: "-:-", rakat: 4 },
  ]);
  const [pref, setPref] = useState({
    location: "Bakı",
    currentPrayer: -1,
    nowis: format(new Date(), "HH:mm"),
    tarix: format(new Date(), "EEEE, d MMMM yyyy", { locale: az }),
    today: getDayOfYear(new Date()),
    prayerLoader: 1,
  });
  const [ayah, setAyah] = useState({
    loader: true,
    content: {
      s: 40,
      a: 60,
      c:
        "Rəbbiniz dedi: 'Mənə dua edin, Mən də sizə cavab verim. Həqiqətən, Mənə ibadət etməyə təkəbbür göstərənlər Cəhənnəmə zəlil olaraq girəcəklər'.",
    },
  });
  const [city, setCity] = useState(parseInt(localStorage.getItem("city")) || 1);
  const [dd, setDd] = useState(parseInt(pref.today));

  const per = (curr, prayers, nowis) => {
    const tmpDate = new Date();
    const untillNow = differenceInSeconds(
      parse(nowis, "HH:mm", tmpDate),
      parse(prayers[curr], "HH:mm", tmpDate)
    );

    // const next = curr === 5 ? 0 : curr + 1;
    let untillNext;
    if (curr === 5) {
      untillNext = differenceInSeconds(
        parse("23:59", "HH:mm", tmpDate),
        parse(prayers[curr], "HH:mm", tmpDate)
      );
    } else {
      untillNext = differenceInSeconds(
        parse(prayers[curr++], "HH:mm", tmpDate),
        parse(prayers[curr], "HH:mm", tmpDate)
      );
    }

    return Math.abs(Math.floor((untillNow * 100) / untillNext));
  };

  const changeCity = (v) => {
    if (!(v in cities)) v = 0;

    localStorage.setItem("city", parseInt(v));

    setPref((prev) => {
      return { ...prev, location: cities[v] };
    });
    setCity(parseInt(v));
  };

  const changeDd = (v) => {
    setPref((prev) => {
      return { ...prev, tarix: prev.tarix };
    });
    setDd(parseInt(v));
  };

  useEffect(() => {
    fetch("https://quran.az/api/random/1/Namaz")
      .then((response) => response.json())
      .then((data) => {
        setAyah({ content: data.out[0], loader: false });
      });

    let tmpCity = parseInt(localStorage.getItem("city"));
    if (tmpCity) {
      setCity(tmpCity);
    }
  }, []);

  useEffect(() => {
    let url = "https://nam.az/api/" + city + "/" + dd;
    fetch(url)
      .then((response) => response.json())
      .then((data) => {
        let currentPrayer = 5;
        const newDate = new Date();

        const tmpPrayers = prayers.map((prayer, i) => {
          prayer["time"] = data.prayers[i];
          prayer["ago"] = formatDistanceStrict(
            newDate,
            parse(data.prayers[i], "HH:mm", newDate),
            { locale: az, addSuffix: true }
          );
          if (data.prayers[i] < pref.nowis) {
            currentPrayer = i;
          }
          return prayer;
        });

        let progress = 0;
        if (pref.today !== data.dd) {
          currentPrayer = -1;
        } else {
          progress = per(currentPrayer, data.prayers, pref.nowis);
        }

        setPrayers([...tmpPrayers]);

        setPref((prev) => {
          return {
            ...prev,
            progress: progress,
            currentPrayer: currentPrayer,
            location: cities[city],
            tarix: data.tarix,
            hijri: data.hijri,
            prayerLoader: 0,
          };
        });
        // console.log("City: " + city + ". Doy: " + dd);
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [city, dd]);

  return (
    <div>
      <NavBar changeCity={changeCity} cities={cities} city={city} />

      <div className="container">
        <Location
          location={pref.location}
          tarix={pref.tarix}
          hijri={pref.hijri}
          dd={dd}
          changeDd={changeDd}
        />
        <Progress bar={pref.progress} />

        {pref.today === dd ? (
          <PrayerList prayers={prayers} currentPrayer={pref.currentPrayer} />
        ) : (
          <PrayerListStill prayers={prayers} />
        )}
        {ayah.loader === true ? <Loader /> : <Ayah ayah={ayah.content} />}
      </div>

      <footer className="footer">
        <nav>
          <ol className="breadcrumb">
            <li className="breadcrumb-item active">&copy; 2020</li>
            <li className="breadcrumb-item">
              <a href="https://www.nam.az">Nam.az</a>
            </li>
            <li className="breadcrumb-item">
              <a href="https://www.quran.az">Quran.az</a>
            </li>
          </ol>
        </nav>
      </footer>
    </div>
  );
};

export default App;
