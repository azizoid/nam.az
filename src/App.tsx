import { useState, useEffect, useRef, lazy, Suspense } from "react";

import { NavBar } from "./components/Navbar/Navbar";
//import Ramadan from "./components/ramadan.component";

import { Progress} from "./components/Progress/Progress";
import { PrayerList } from "./components/PrayerList/PrayerList";
import { PrayerListStill } from "./components/PrayerList/PrayerListStill";

import { Footer } from "./components/Footer/Footer";
import { Loader } from "./components/Loader/Loader";

import {
  format,
  formatDistanceStrict,
  parse,
  differenceInSeconds,
  getDayOfYear,
} from "date-fns";
import az from "date-fns/locale/az";

import cities from "./assist/cities";

import { PrayerProps } from './components/PrayerList/Prayer'

import styles from './App.module.scss'
import classNames from "classnames";

const Location = lazy(() => import("./components/Location/Location"));
const Ayah = lazy(() => import("./components/Ayah/Ayah"));

const App = ():JSX.Element => {
  const newDate = useRef(new Date());
  const today = getDayOfYear(newDate.current) + 1;

  const [prayers, setPrayers] = useState([
    { id: 1, time: "-:-", rakat: 2, ago: "", title: "Sübh namazı" },
    { id: 2, time: "-:-", rakat: 0, ago: "", title: "Günəş" },
    { id: 3, time: "-:-", rakat: 4, ago: "", title: "Zöhr namazı" },
    { id: 4, time: "-:-", rakat: 4, ago: "", title: "Əsr namazı" },
    { id: 5, time: "-:-", rakat: 3, ago: "", title: "Məğrib namazı" },
    { id: 6, time: "-:-", rakat: 4, ago: "", title: "İşa namazı" },
  ]);

  const [pref, setPref] = useState({
    location: "Bakı",
    currentPrayer: -1,
    nowis: format(newDate.current, "HH:mm"),
    tarix: format(newDate.current, "EEEE, d MMMM yyyy", { locale: az }),
    hijri: "",
    today: today,
    progress: 0,
    ramadan: 0,
  });

  const [city, setCity] = useState(
    JSON.parse(localStorage.getItem("city") as string) || "1"
  );
  const [dd, setDd] = useState(today);

  useEffect(() => {
    const fetchData = async () => {
      
      await fetch(`https://nam.az/api/${city}/${dd}`)
        .then((response) => response.json())
        .then((data:any) => {
          let currentPrayer = 5;

          setPrayers((prev) =>
            prev.map((prayer: PrayerProps, i) => {

              prayer["time"] = data.prayers[i];

              prayer["ago"] = formatDistanceStrict(
                newDate.current,
                parse(data.prayers[i], "HH:mm", newDate.current),
                { locale: az, addSuffix: true }
              );

              if (data.prayers[i] < pref.nowis) {
                currentPrayer = i;
              }

              return prayer;
            })
          );

          let progress = 0;

          if (pref.today !== data.dd) {
            currentPrayer = -1;
          } else {
            progress = percentageCounter(
              currentPrayer,
              data.prayers,
              pref.nowis
            );
          }

          setPref((prev) => ({
            ...prev,
            progress: progress,
            currentPrayer: currentPrayer,
            location: cities[city],
            tarix: data.tarix,
            hijri: data.hijri,
            ramadan: data.dd - 103,
          }));
        });
    };
    fetchData();
  }, [city, dd, pref.nowis, pref.today]);

  const percentageCounter = (
    currentPrayer: number,
    prayersFromApi: string[],
    nowis: string
  ): number => {

    const untillNow = differenceInSeconds(
      parse(nowis, "HH:mm", newDate.current),
      parse(prayersFromApi[currentPrayer], "HH:mm", newDate.current)
    );

    let untillNext;
    
    if (currentPrayer === 5) {
      untillNext = differenceInSeconds(
        parse("23:59", "HH:mm", newDate.current),
        parse(prayersFromApi[currentPrayer], "HH:mm", newDate.current)
      );
    } else {
      untillNext = differenceInSeconds(
        parse(prayersFromApi[currentPrayer++], "HH:mm", newDate.current),
        parse(prayersFromApi[currentPrayer], "HH:mm", newDate.current)
      );
    }

    return Math.abs(Math.floor((untillNow * 100) / untillNext));
  };

  const changeCity = (v: number): void => {
    if (!(v in cities)) return;

    if (v !== 0) {
      localStorage.setItem("city", JSON.stringify(v));
      setPref((prev) => ({ ...prev, location: cities[v] }));
      setCity(v);
    }
  };

  return (
    <>
      <NavBar changeCity={changeCity} city={city} />

      <div className={classNames( "d-flex flex-column align-items-between justify-content-sm-start justify-content-md-between", styles.mainBlock,)}>
          <Suspense fallback={<Loader />}>
            <Location
              location={pref.location}
              tarix={pref.tarix}
              hijri={pref.hijri}
              dd={dd}
              changeDd={(v: number) => setDd(v)}
            />
            
            <Progress bar={pref.progress} />
          </Suspense>

          {pref.today === dd ? (
            <PrayerList prayers={prayers} currentPrayer={pref.currentPrayer} />
          ) : (
            <PrayerListStill prayers={prayers} />
          )}

          <Suspense fallback={<Loader />}>
            <Ayah />
          </Suspense>

          <Footer />    
      </div>
      
      
      
    </>
  );
};

export default App;
