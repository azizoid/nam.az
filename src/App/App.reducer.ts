import { format, getDayOfYear, isLeapYear } from 'date-fns';
import { selectCity, useLocalStorage } from 'utility';

const newDate = new Date();
const today = getDayOfYear(newDate) + (isLeapYear(newDate) ? 0 : 1);

const [readCity, writeCity] = useLocalStorage<number>('city', 0);

type StateProps = {
  city: number;
  location: string;
  currentPrayer: number;
  nowis: string;
  tarix: string;
  hijri: string;
  today: number;
  progress: number;
  ramadan: number;
  prayers: ApiPrayerProps[];
};

type ApiPrayerProps = {
  id: number;
  time: string;
  rakat: number;
  ago: string;
  title: string;
};

export const AppInitialState: StateProps = {
  city: readCity,
  location: '',
  currentPrayer: 5,
  nowis: format(newDate, 'HH:mm'),
  tarix: '',
  hijri: '',
  today: today,
  progress: 0,
  ramadan: 0,
  prayers: [
    { id: 1, time: '-:-', rakat: 2, ago: '', title: 'Sübh namazı' },
    { id: 2, time: '-:-', rakat: 0, ago: '', title: 'Gün çıxır' },
    { id: 3, time: '-:-', rakat: 4, ago: '', title: 'Zöhr namazı' },
    { id: 4, time: '-:-', rakat: 4, ago: '', title: 'Əsr namazı' },
    { id: 5, time: '-:-', rakat: 3, ago: '', title: 'Məğrib namazı' },
    { id: 6, time: '-:-', rakat: 4, ago: '', title: 'İşa namazı' },
  ],
};

type ActionProps = {
  type: 'init' | 'location' | 'dayOfTheYear';
  payload: Partial<StateProps> | string | number;
};

// The reducer function
export const AppReducer = (state: StateProps, action: ActionProps) => {
  switch (action.type) {
    case 'init':
      return {
        ...state,
        ...(typeof action.payload === 'object' && action.payload),
      };
    case 'location':
      const cityToNumber = Number(action.payload);

      writeCity(cityToNumber);

      const changeCityTo = selectCity(cityToNumber);

      return {
        ...state,
        location: String(changeCityTo),
        city: cityToNumber,
        today,
      };
    case 'dayOfTheYear':
      return { ...state, today: Number(action.payload) };

    default:
      throw new Error();
  }
};
