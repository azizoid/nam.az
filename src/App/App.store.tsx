import { createContext, Dispatch } from 'react';
import { ActionProps, AppInitialState, StateProps } from './App.reducer';

export type MyContextValue = {
  state: StateProps;
  dispatch: Dispatch<ActionProps>;
};

export const MyContext = createContext<MyContextValue>({
  state: AppInitialState,
  dispatch: () => {},
});
