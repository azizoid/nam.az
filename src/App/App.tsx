import { lazy, Suspense, useEffect, useReducer, useState } from 'react';
import {
  fetchData,
  percentageCounter,
  ResponseDataProps,
  selectCity,
  useLocalStorage,
} from 'utility';

import { Footer, Header, Loader } from 'ui';

import {
  formatDistanceStrict,
  getDayOfYear,
  isLeapYear,
  parse,
} from 'date-fns';

import az from 'date-fns/locale/az';

import { Location } from './Location/Location';
import { Progress } from './Progress/Progress';
import type { PrayerProps } from './PrayerList/Prayer';
import { PrayerList } from './PrayerList/PrayerList';
import { PrayerListStill } from './PrayerList/PrayerListStill';
import { AppInitialState, AppReducer } from './App.reducer';

const Ayah = lazy(() => import('./Ayah/Ayah'));

const newDate = new Date();
const today = getDayOfYear(newDate) + (isLeapYear(newDate) ? 0 : 1);

export const App = () => {
  const [localStorageCity, setLocalStorageCity] = useLocalStorage<
    number | undefined
  >('city', undefined);

  const [state, dispatch] = useReducer(AppReducer, AppInitialState);

  const [prayers, setPrayers] = useState([
    { id: 1, time: '-:-', rakat: 2, ago: '', title: 'Sübh namazı' },
    { id: 2, time: '-:-', rakat: 0, ago: '', title: 'Gün çıxır' },
    { id: 3, time: '-:-', rakat: 4, ago: '', title: 'Zöhr namazı' },
    { id: 4, time: '-:-', rakat: 4, ago: '', title: 'Əsr namazı' },
    { id: 5, time: '-:-', rakat: 3, ago: '', title: 'Məğrib namazı' },
    { id: 6, time: '-:-', rakat: 4, ago: '', title: 'İşa namazı' },
  ]);

  useEffect(() => {
    if (state.city) {
      setLocalStorageCity(state.city);
    }
  }, [state.city]);

  useEffect(() => {
    if (!localStorageCity) {
      return;
    }

    fetchData({ city: localStorageCity, dd: state.today })
      .then((data: ResponseDataProps) => {
        let currentPrayer = 5;

        setPrayers(prev =>
          prev.map((prayer: PrayerProps, i) => {
            prayer.time = data.prayers[i];

            prayer.ago = formatDistanceStrict(
              newDate,
              parse(data.prayers[i], 'HH:mm', newDate),
              { locale: az, addSuffix: true }
            );

            if (data.prayers[i] < state.nowis) {
              currentPrayer = i;
            }

            return prayer;
          })
        );

        let progress = 0;

        if (today !== data.dd) {
          currentPrayer = -1;
        } else {
          progress = percentageCounter({
            currentPrayer,
            apiPrayers: data.prayers,
            nowis: state.nowis,
            newDate,
          });
        }

        dispatch({
          type: 'init',
          payload: {
            progress,
            currentPrayer,
            location: selectCity(data.city),
            tarix: data.tarix,
            hijri: data.hijri,
          },
        });
      })
      .catch(error => {
        // console.error(error);
      });
  }, [localStorageCity, state.today]);

  const changeCity = (v: number): void => {
    dispatch({ type: 'location', payload: v });
  };

  // useEffect(() => {
  //   console.table(state);
  // }, [state]);

  return (
    <div className="min-h-screen flex flex-col justify-between">
      <Header changeCity={changeCity} city={localStorageCity} />

      <div className="align-middle container mx-auto my-10 pb-2">
        <Suspense fallback={<Loader />}>
          <Location
            location={state.location}
            tarix={state.tarix}
            hijri={state.hijri}
            dd={state.today}
            changeDd={(day: number) =>
              dispatch({ type: 'dayOfTheYear', payload: day })
            }
          />

          <Progress bar={state.progress} />
        </Suspense>

        {state.today === today ? (
          <PrayerList
            prayers={prayers}
            currentPrayer={state.currentPrayer}
            progress={state.progress}
          />
        ) : (
          <PrayerListStill prayers={prayers} />
        )}

        <Suspense fallback={<Loader />}>
          <Ayah />
        </Suspense>
      </div>

      <Footer />
    </div>
  );
};
