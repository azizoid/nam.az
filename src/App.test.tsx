import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import App from './App';

test('renders App', async () => {
  const container = render(<App />);

  expect(container).toMatchSnapshot();
});
