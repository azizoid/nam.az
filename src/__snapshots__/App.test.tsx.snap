// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar sticky-top navbar-expand-lg navbar-light bg-light"
      >
        <div
          class="container"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            <img
              alt="nam.az"
              class="d-inline-block align-top"
              height="30"
              src="/favicon.png"
              width="30"
            />
            Nam.az
          </a>
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item active"
            >
              <select
                aria-label="Haradasınız?"
                class="form-select btn-outline-success"
              >
                <option
                  value="0"
                >
                  Haradasınız?
                </option>
                <option
                  value="1"
                >
                  Bakı
                </option>
                <option
                  value="2"
                >
                  Sumqayıt
                </option>
                <option
                  value="3"
                >
                  Gəncə
                </option>
                <option
                  value="4"
                >
                  Lənkaran
                </option>
                <option
                  value="5"
                >
                  Mingəçevir
                </option>
                <option
                  value="6"
                >
                  Naxçıvan
                </option>
                <option
                  value="7"
                >
                  Ağdam
                </option>
                <option
                  value="8"
                >
                  Astara
                </option>
                <option
                  value="9"
                >
                  Beyləqan
                </option>
                <option
                  value="10"
                >
                  Cəlilabad
                </option>
                <option
                  value="11"
                >
                  Göyçay
                </option>
                <option
                  value="12"
                >
                  Qəbələ
                </option>
                <option
                  value="13"
                >
                  Qazax
                </option>
                <option
                  value="14"
                >
                  Quba
                </option>
                <option
                  value="15"
                >
                  Qusar
                </option>
                <option
                  value="16"
                >
                  Saatlı
                </option>
                <option
                  value="17"
                >
                  Sabirabad
                </option>
                <option
                  value="18"
                >
                  Şamaxı
                </option>
                <option
                  value="19"
                >
                  Şəki
                </option>
                <option
                  value="20"
                >
                  Şirvan
                </option>
                <option
                  value="21"
                >
                  Xaçmaz
                </option>
                <option
                  value="22"
                >
                  Yevlax
                </option>
                <option
                  value="23"
                >
                  Zaqatala
                </option>
                <option
                  value="24"
                >
                  Şuşa
                </option>
                <option
                  value="25"
                >
                  Marneuli
                </option>
                <option
                  value="26"
                >
                  Bolnisi
                </option>
              </select>
              <small>
                Bakı, Gəncə, 
                <u>
                  Şuşa
                </u>
                 və digər
              </small>
            </li>
          </ul>
        </div>
      </nav>
      <div
        class="d-flex flex-column align-items-between justify-content-sm-start justify-content-md-between mainBlock"
      >
        <div
          class="progress"
          style="display: none;"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            class="progress bg-info"
            role="progressbar"
            style="width: 0px;"
          />
        </div>
        <div
          class="text-center"
        >
          <div
            class="ldsFacebook"
          >
            <div />
            <div />
            <div />
          </div>
        </div>
        <div
          class="row times"
        >
          <div
            class="col-12 col-md-4 alert"
          >
            <div
              class="row"
            >
              <h6
                class="col-7 col-md-12 align-self-center"
              >
                Sübh namazı
              </h6>
              <div
                class="col-5 col-md-12"
              >
                <h4>
                  -:-
                </h4>
                <small />
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-4 alert text-muted"
          >
            <div
              class="row"
            >
              <h6
                class="col-7 col-md-12 align-self-center"
              >
                Günəş
              </h6>
              <div
                class="col-5 col-md-12"
              >
                <h4>
                  -:-
                </h4>
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(246, 153, 63);"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-4 alert"
          >
            <div
              class="row"
            >
              <h6
                class="col-7 col-md-12 align-self-center"
              >
                Zöhr namazı
              </h6>
              <div
                class="col-5 col-md-12"
              >
                <h4>
                  -:-
                </h4>
                <small />
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-4 alert"
          >
            <div
              class="row"
            >
              <h6
                class="col-7 col-md-12 align-self-center"
              >
                Əsr namazı
              </h6>
              <div
                class="col-5 col-md-12"
              >
                <h4>
                  -:-
                </h4>
                <small />
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-4 alert"
          >
            <div
              class="row"
            >
              <h6
                class="col-7 col-md-12 align-self-center"
              >
                Məğrib namazı
              </h6>
              <div
                class="col-5 col-md-12"
              >
                <h4>
                  -:-
                </h4>
                <small />
              </div>
            </div>
          </div>
          <div
            class="col-12 col-md-4 alert"
          >
            <div
              class="row"
            >
              <h6
                class="col-7 col-md-12 align-self-center"
              >
                İşa namazı
              </h6>
              <div
                class="col-5 col-md-12"
              >
                <h4>
                  -:-
                </h4>
                <small />
              </div>
            </div>
          </div>
        </div>
        <div
          class="text-center"
        >
          <div
            class="ldsFacebook"
          >
            <div />
            <div />
            <div />
          </div>
        </div>
        <nav
          class="navbar "
        >
          <div
            class="container-fluid"
          >
            <ul
              class="nav align-content-center"
            >
              <li
                class="nav-item navbar-text active"
              >
                © 
                2021
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="https://www.nam.az"
                >
                  Nam.az
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="https://www.quran.az"
                  rel=" noopener noreferrer"
                  target="_blank"
                >
                  Quran.az
                </a>
              </li>
            </ul>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar sticky-top navbar-expand-lg navbar-light bg-light"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="nam.az"
            class="d-inline-block align-top"
            height="30"
            src="/favicon.png"
            width="30"
          />
          Nam.az
        </a>
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item active"
          >
            <select
              aria-label="Haradasınız?"
              class="form-select btn-outline-success"
            >
              <option
                value="0"
              >
                Haradasınız?
              </option>
              <option
                value="1"
              >
                Bakı
              </option>
              <option
                value="2"
              >
                Sumqayıt
              </option>
              <option
                value="3"
              >
                Gəncə
              </option>
              <option
                value="4"
              >
                Lənkaran
              </option>
              <option
                value="5"
              >
                Mingəçevir
              </option>
              <option
                value="6"
              >
                Naxçıvan
              </option>
              <option
                value="7"
              >
                Ağdam
              </option>
              <option
                value="8"
              >
                Astara
              </option>
              <option
                value="9"
              >
                Beyləqan
              </option>
              <option
                value="10"
              >
                Cəlilabad
              </option>
              <option
                value="11"
              >
                Göyçay
              </option>
              <option
                value="12"
              >
                Qəbələ
              </option>
              <option
                value="13"
              >
                Qazax
              </option>
              <option
                value="14"
              >
                Quba
              </option>
              <option
                value="15"
              >
                Qusar
              </option>
              <option
                value="16"
              >
                Saatlı
              </option>
              <option
                value="17"
              >
                Sabirabad
              </option>
              <option
                value="18"
              >
                Şamaxı
              </option>
              <option
                value="19"
              >
                Şəki
              </option>
              <option
                value="20"
              >
                Şirvan
              </option>
              <option
                value="21"
              >
                Xaçmaz
              </option>
              <option
                value="22"
              >
                Yevlax
              </option>
              <option
                value="23"
              >
                Zaqatala
              </option>
              <option
                value="24"
              >
                Şuşa
              </option>
              <option
                value="25"
              >
                Marneuli
              </option>
              <option
                value="26"
              >
                Bolnisi
              </option>
            </select>
            <small>
              Bakı, Gəncə, 
              <u>
                Şuşa
              </u>
               və digər
            </small>
          </li>
        </ul>
      </div>
    </nav>
    <div
      class="d-flex flex-column align-items-between justify-content-sm-start justify-content-md-between mainBlock"
    >
      <div
        class="progress"
        style="display: none;"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          class="progress bg-info"
          role="progressbar"
          style="width: 0px;"
        />
      </div>
      <div
        class="text-center"
      >
        <div
          class="ldsFacebook"
        >
          <div />
          <div />
          <div />
        </div>
      </div>
      <div
        class="row times"
      >
        <div
          class="col-12 col-md-4 alert"
        >
          <div
            class="row"
          >
            <h6
              class="col-7 col-md-12 align-self-center"
            >
              Sübh namazı
            </h6>
            <div
              class="col-5 col-md-12"
            >
              <h4>
                -:-
              </h4>
              <small />
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-4 alert text-muted"
        >
          <div
            class="row"
          >
            <h6
              class="col-7 col-md-12 align-self-center"
            >
              Günəş
            </h6>
            <div
              class="col-5 col-md-12"
            >
              <h4>
                -:-
              </h4>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                style="color: rgb(246, 153, 63);"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-4 alert"
        >
          <div
            class="row"
          >
            <h6
              class="col-7 col-md-12 align-self-center"
            >
              Zöhr namazı
            </h6>
            <div
              class="col-5 col-md-12"
            >
              <h4>
                -:-
              </h4>
              <small />
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-4 alert"
        >
          <div
            class="row"
          >
            <h6
              class="col-7 col-md-12 align-self-center"
            >
              Əsr namazı
            </h6>
            <div
              class="col-5 col-md-12"
            >
              <h4>
                -:-
              </h4>
              <small />
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-4 alert"
        >
          <div
            class="row"
          >
            <h6
              class="col-7 col-md-12 align-self-center"
            >
              Məğrib namazı
            </h6>
            <div
              class="col-5 col-md-12"
            >
              <h4>
                -:-
              </h4>
              <small />
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-4 alert"
        >
          <div
            class="row"
          >
            <h6
              class="col-7 col-md-12 align-self-center"
            >
              İşa namazı
            </h6>
            <div
              class="col-5 col-md-12"
            >
              <h4>
                -:-
              </h4>
              <small />
            </div>
          </div>
        </div>
      </div>
      <div
        class="text-center"
      >
        <div
          class="ldsFacebook"
        >
          <div />
          <div />
          <div />
        </div>
      </div>
      <nav
        class="navbar "
      >
        <div
          class="container-fluid"
        >
          <ul
            class="nav align-content-center"
          >
            <li
              class="nav-item navbar-text active"
            >
              © 
              2021
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="https://www.nam.az"
              >
                Nam.az
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="https://www.quran.az"
                rel=" noopener noreferrer"
                target="_blank"
              >
                Quran.az
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
